granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -88,13 +197,39 @@ def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> O
     v2_path = find_v2_hierarchy(resolve_host_root_links)
     if v1_paths and controller in v1_paths:
         # Either v1 or hybrid with the requested controller bound to a v1 hierarchy
-        return v1_paths[controller]
+        return CgroupCoreV1(Path(v1_paths[controller]))
     if v2_path:
-        # v2 (unified) - all controllers are in a single unified hierarchy
-        return v2_path
+        # v2 (unified) - check given controller is supported in cgroup v2
+        cgroup_v2 = CgroupCoreV2(Path(v2_path))
+        if cgroup_v2.is_controller_supported(controller):
+            return cgroup_v2
     # No cgroup mount of the requested controller
     return None
 
 
-def is_known_controller(controller: str) -> bool:
-    return controller in SUBSYSTEMS
+def create_cgroup_from_path(controller: str, cgroup_path: Path) -> CgroupCore:
+    cgroup_mount = get_cgroup_mount(controller)
+    assert cgroup_mount is not None, "Could not find cgroup mount point"
```

Comment:
This assert repeats multiple times, move to `get_cgroup_mount` or if you want a variant that can fail, create a `get_cgroup_mount_checked`

Also, add the `controller` to the assert message